packages/cozy-harvest-lib/src/helpers/konnectorBlock.js

```diff
@@ -0,0 +1,227 @@
+import { triggers } from 'cozy-client/dist/models/trigger'
+import { generateUniversalLink } from 'cozy-ui/transpiled/react/AppLinker'
+import get from 'lodash/get'
+
+import {
+  buildKonnectorQuery,
+  buildIsAccountConnectedQuery,
+  buildIsInErrorTriggersQuery,
+  buildTriggersByIdQuery
+} from './queries'
+import { hasNewVersionAvailable } from './konnectors'
+
+const formatKonnector = ({
+  client,
+  konnector,
+  universalLink,
+  iconStatus,
+  message
+}) => {
+  const link = generateUniversalLink({
+    cozyUrl: client.getStackClient().uri,
+    slug: universalLink.slug,
+    subDomainType: client.getInstanceOptions().cozySubdomainType,
+    nativePath: universalLink.nativePath
+  })
+
+  return {
+    name: konnector.name,
+    link,
+    vendorLink: konnector.vendor_link && {
+      component: 'a',
+      href: konnector.vendor_link,
+      target: '_blank'
+    },
+    iconStatus,
+    message
+  }
+}
+
+const getDataFromState = async (client, query) => {
+  try {
+    await client.query(query.definition, query.options)
+    const { data } = client.getQueryFromState(query.options.as)
+    return data[0]
+  } catch (error) {
+    throw error
+  }
+}
```

Comment:
1. It seems to do the same thing as useQuery ? Why not using useQuery directly ?
2. If the function fetches something via a query, it is a bad practice to call the function `get` as it implies that the data fetching is synchronous.

Reply:
`useQuery` n'est utilisé nulle part car ce ne sont ni des hook ni des composants React. Je n'ai pas souhaité utilisé `useQuery` également parce qu'il y a plusieurs requête à faire avant de retourner la réponse finale, et au lieu de les faire en parallèle j'ai préféré les faire en série, afin de faire le moins de requête possible si ce n'est pas nécessaire.

Ok pour le `get` qui va devenir `fetch`